DataFormats/PatCandidates/src/CovarianceParameterization.cc

```diff
@@ -0,0 +1,214 @@
+          break;
+      
+    }
+  return 0;
+}
+float CompressionElement::unpack(uint16_t packed, float ref) const
+{
+    float unpacked=0;
+    switch(method) {
+        case(float16):
+          unpacked= MiniFloatConverter::float16to32(packed)/params[0];
+          break;
+        case(reduceMantissa):
+          unpacked=packed;
+          break;
+        case(logPack):
+          unpacked=logintpack::unpack16log(* reinterpret_cast<int16_t *>(&packed),params[0],params[1],bits);
+          break;
+        case(zero):
+          unpacked=0;
+          break;
+        case(one):
+        case(tanLogPack):
+          unpacked=1; //FIXME: should be implemented
+    }
+    switch(target) {
+        case(realValue):
+          return unpacked;
+        case(ratioToRef):
+          return unpacked*ref;
+        case(differenceToRef):
+          return unpacked+ref;
+    }
+
+    return ref;
+  
+}
+
+
+
+
+
+
+void CovarianceParameterization::load(int version)
+{
+ edm::FileInPath fip((boost::format("DataFormats/PatCandidates/data/CovarianceParameterization_version%d.root") % version).str());
+ fileToRead_ = TFile::Open(fip.fullPath().c_str()); 
+ TFile & fileToRead=*fileToRead_;
+//Read files from here fip.fullPath().c_str();
+ if(fileToRead.IsOpen())  {
+     readFile(fileToRead);
+
+     TIter next(((TDirectoryFile*)fileToRead.Get("schemas"))->GetListOfKeys());
+     TKey *key;
+     while ((key = (TKey*)next())) {
+      TClass *cl = gROOT->GetClass(key->GetClassName());
+      if (!cl->InheritsFrom("TDirectoryFile")) continue;
+      std::string schemaNumber = key->ReadObj()->GetName();
+      uint16_t schemaN = std::stoi(schemaNumber);
+     //for (int folderNumber = 0; folderNumber < 6 ; folderNumber++) {
+      CompressionSchema schema; 
+      for (int i = 0; i < 5; i++) {
+            for (int j = i; j < 5; j++) {        //FILLING ONLY THE SCHEMA OF SOME ELEMENTS
+                  std::string folder = "schemas/" + schemaNumber + "/"  + char(48+i) + char(48+j);
+                std::string methodString = folder + "/method";
+                std::string targetString = folder + "/target";
+                std::string bitString = folder + "/bit";
+                std::vector<float> vParams ;
+                TVector *p=(TVector*) fileToRead.Get((folder+"/param").c_str());
+                for(int k = 0 ; k < p->GetNoElements() ; k++){
+                    vParams.push_back((*p)[k]);
+                }
+
+                schema(i,j)=CompressionElement((CompressionElement::Method) ((TParameter<int>*) fileToRead.Get(methodString.c_str()))->GetVal(),
+		(CompressionElement::Target) ((TParameter<int>*) fileToRead.Get(targetString.c_str()))->GetVal(),
+		 (int) ((TParameter<int>*) fileToRead.Get(bitString.c_str()))->GetVal(), vParams);
+        
+                
+            }
+        }
+     schemas[schemaN]=schema; 
+     }
+
+    loadedVersion_=version; 
+ } else {loadedVersion_=-1;}
+
+}
```

Comment:
can the TFile be deleted here? (it needs to be deleted somewhere - the sooner the better)
@arizzi